ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -1740,6 +1740,10 @@ static void ggml_vk_load_shaders(vk_device& device) {
         m_warptile_mmq_int = { 128,  64,  64, 32, subgroup_size_8,     32, 2, 2, 2, 1, subgroup_size_8 };
         s_warptile_mmq_int = { subgroup_size_32, 32, 32, 32, 32,       32, 2, 2, 1, 1, subgroup_size_8 };
 
+        // chip specific tuning
+        if (device->architecture == AMD_GCN)
+            m_warptile_mmq = { 256, 64, 64, 32, 16, 16, 2, 2, tn_m, tk_m, 16 };
```

Comment:
```suggestion
        if (device->architecture == AMD_GCN) {
            m_warptile_mmq = m_warptile_mmq_int = { 256, 64, 64, 32, 16, 16, 2, 2, 2, 1, 16 };
        }
```

Interesting that this increases performance, as it breaks up each warp/wave of 64 into 4 blocks of 16. I don't see a reason to keep the tn/tk variables. You'd need all three for them to work with coopmat, but no GCN GPU supports it anyways.

I checked and your tune also works for integer dot on Vega20.

Reply:
> Interesting that this increases performance, as it breaks up each warp/wave of 64 into 4 blocks of 16.

Yeah the strangest part here is that it while it provides the same tokens per second per mhz it lets the chip clock higher before hitting the TDC limit. It could be just be something as simple as the compiler optimizing things differently but in any case it's running faster than before.